06_timers/kernel_space/time.c

```diff
@@ -0,0 +1,71 @@
+// SPDX-License-Identifier: GPL-2.0
+#include <asm/uaccess.h>
+#include <linux/cdev.h>
+#include <linux/device.h>
+#include <linux/fs.h>
+#include <linux/init.h>
+#include <linux/module.h>
+#include <linux/pci.h>
+#include <linux/version.h>
+#include <linux/ktime.h>
+
+MODULE_DESCRIPTION("Basic module demo: work with time");
+MODULE_AUTHOR("Anton KOtsiubailo");
+MODULE_VERSION("0.1");
+MODULE_LICENSE("Dual MIT/GPL"); // this affects the kernel behavior
+
+#define LEN_MSG 160
+#define CLASS_ATTR(_name, _mode, _show, _store)				\
+struct class_attribute class_attr_##_name =				\
+__ATTR(_name, _mode, _show, _store)
+
+static char buf_msg[LEN_MSG + 1] = "Hello from module!\n";
+static ktime_t currtime;
+static ktime_t pasttime;
+static ktime_t prev;
+
+static ssize_t x_show(struct class *class, struct class_attribute *attr,
+			char *buf)
+{
+	struct timespec time;
+
+	currtime = ktime_get();
+	time = ktime_to_timespec(currtime - pasttime);
+	strcpy(buf, buf_msg);
+	pr_info("read %ld\n", (long)strlen(buf));
+	pr_info("time: %d.%09ld sec\n", time.tv_sec, time.tv_nsec);
+	time = ktime_to_timespec(prev);
+	pr_info("previous time: %d.%09ld sec\n", time.tv_sec, time.tv_nsec);
+	prev = currtime - pasttime;
```

Comment:
previous time displayed incorrectly
according to your code you need to print pasttime variable
prev variable is not needed